ansible/roles/postgresql/tasks/backup.yml

```diff
@@ -0,0 +1,20 @@
+- name: create backup directory
+  become: yes
+  file:
+    path: '{{ postgresql_backup_dir }}'
+    owner: postgres
+    group: postgres
+    mode: 0755
+    state: directory
+
+- include_tasks: backup_plain.yml
+  when: backup_postgres == 'plain' and (is_pg_standby or not hot_standby_server|default(False))
```

Comment:
@snopoke this line seems to be failing for my quick aws env setup (which I'm doing without a `pg_standby`). Can you explain this line to me / when this is supposed to run?

My understanding is `hot_standby_server` is only ever set as a variable on a `pg_standby` host, so if `is_pg_standby` is false, then `hot_standby_server` is undefined, `hot_standby_server|default(False))` is `False`, and `not hot_standby_server|default(False))` is `True`—so `(is_pg_standby or not hot_standby_server|default(False))` is always `True`. Is something in this logic inverted, or am I maybe missing something?

Reply:
This will be true whenever the current host is a standby (is_pg_standby) or no standby is defined.  I will not be true when running on a master with no standby.

hot_standby_server is only set on a master (not a slave as you said), hot_standby_master is set on the slave.

Reply:
I will work on creating a pg_backup group. to get rid of the "standby has to be backup server" constraint.  Does that sound like a good solution to you?

Reply:
As @javierwilson pointed out the `hot_standby_server` var is only set on the master, not the standby node. So the logic here is:
* if backup is 'on' and
  * we're on a standby node
  * or there is no standby node configured

Reply:
OK thanks, that helps, I should be able to debug from here. Sounds like part of what I was missing is that even if you don't have a `pg_standby`, we still do this backup, just on the same machine? (The other part was that I was confusing the variables `hot_standby_server` and `hot_standby_master`.)